controlplane/kubeadm/api/v1beta1/v1beta2_condition_consts.go

```diff
@@ -0,0 +1,75 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+
+// Conditions that will be used for the KubeadmControlPlane object in v1Beta2 API version.
+const (
+	// KubeadmControlPlaneAvailableV1beta2Condition True if the control plane can be reached, EtcdClusterAvailable is true,
+	// and CertificatesAvailable is true.
+	KubeadmControlPlaneAvailableV1beta2Condition = clusterv1.AvailableV1beta2Condition
+
+	// KubeadmControlPlaneCertificatesAvailableV1beta2Condition True if all the cluster certificates exist.
+	KubeadmControlPlaneCertificatesAvailableV1beta2Condition = "CertificatesAvailable"
+
+	// KubeadmControlPlaneEtcdClusterAvailableV1beta2Condition surfaces issues to the managed etcd cluster, if any.
+	// It is computed as aggregation of Machines's EtcdMemberHealthy (if not using an external etcd) conditions plus
+	// additional checks validating potential issues to etcd quorum.
+	KubeadmControlPlaneEtcdClusterAvailableV1beta2Condition = "EtcdClusterAvailable"
+
+	// KubeadmControlPlaneMachinesReadyV1beta2Condition surfaces detail of issues on the controlled machines, if any.
+	// Please note this will include also APIServerPodHealthy, ControllerManagerPodHealthy, SchedulerPodHealthy conditions.
```

Comment:
Did we think through the impact that this will have on rollout behavior etc.? (or are we already checking for these conditions today)

(sorry don't remember previous discussion results)

Reply:
There is still some details to be figured out, e.g. Ready also includes NodeReady, which implies CNI (which is not checked today)